src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -134,6 +138,11 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
     },
 
     updateContainer(element : ReactNodeList, container : OpaqueNode, parentComponent : ?ReactComponent<any, any, any>, callback: ?Function) : void {
+      invariant(
+        !ReactFeatureFlags.disableNewFiberFeatures || isValidElement(element),
+        'render(): Invalid component element.'
```

Comment:
Can you make sure null/false works? And we might as well use the old error message:

```
      invariant(
        inst === null ||
        inst === false ||
        React.isValidElement(inst),
        '%s(...): A valid React element (or null) must be returned. You may have ' +
        'returned undefined, an array or some other invalid object.',
        Component.displayName || Component.name || 'Component'
      );
```